### PR TITLE
Added API to return user count bands and user count

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Added API to return user count bands and user count

#### Ticket Link
Fixes https://mattermost.atlassian.net/jira/software/c/projects/MM/boards/126?assignee=6054be09e394c30069d776ea&selectedIssue=MM-56351

#### Release Note
```release-note
None
```

### CODE REVIEW

File: server/channels/api4/limits.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/mattermost/mattermost/server/public/model"
+
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+)
+
+func (api *API) InitLimits() {
+	api.BaseRoutes.Limits.Handle("/users", api.APISessionRequired(getUserLimits)).Methods("GET")
+}
+
+func getUserLimits(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (larkox) [inline server/channels/api4/limits.go:19]: 0/5 do we want to add tests at api level?

File: server/channels/app/limits.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+func (a *App) shouldShowUserLimits() bool {
+	license := a.License()
+
+	if license == nil {
+		return true
+	}
+
+	if maxUsersLimit == 0 {
+		return false
+	}
```
Comment (larkox) [inline server/channels/app/limits.go:45]: Shouldn't these conditions be the other way around? Check first the max users limit before checking the license?

That, or change the first condition to `!= nil return false` and return true at the end.

File: server/channels/app/limits_test.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+	t.Run("base case", func(t *testing.T) {
+		userLimits, appErr := th.App.GetUserLimits()
+		require.Nil(t, appErr)
+
+		// InitBasic creates 3 users by default
+		require.Equal(t, int64(3), userLimits.ActiveUserCount)
+		require.Equal(t, int64(10000), userLimits.MaxUsersLimit)
+		require.Equal(t, int64(500), userLimits.LowerBandUserLimit)
+		require.Equal(t, int64(9000), userLimits.UpperBandUserLimit)
```
Comment (larkox) [inline server/channels/app/limits_test.go:25]: These are a bit brittle when we change the values of the constants. We may want to use the constants, but 0/5.

File: server/channels/app/limits_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	})
+
+	t.Run("user count should increase on creating new user and decrease on deleting", func(t *testing.T) {
+		userLimits, appErr := th.App.GetUserLimits()
+		require.Nil(t, appErr)
+		require.Equal(t, int64(3), userLimits.ActiveUserCount)
```
Comment (larkox) [inline server/channels/app/limits_test.go:31]: This initial value of 3 sounds a bit brittle, but probably is fine.

File: server/channels/app/limits_test.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		require.Nil(t, appErr)
+
+		// InitBasic creates 3 users by default
+		require.Equal(t, int64(3), userLimits.ActiveUserCount)
+		require.Equal(t, int64(10000), userLimits.MaxUsersLimit)
+		require.Equal(t, int64(500), userLimits.LowerBandUserLimit)
+		require.Equal(t, int64(9000), userLimits.UpperBandUserLimit)
+	})
+
+	t.Run("user count should increase on creating new user and decrease on deleting", func(t *testing.T) {
```
Comment (larkox) [inline server/channels/app/limits_test.go:28]: Just to be thorough, we may consider adding either a new case, or to this case, guest users.

File: server/channels/app/limits_test.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+		// now we create a new user
+		newUser := th.CreateUser()
+
+		userLimits, appErr = th.App.GetUserLimits()
+		require.Nil(t, appErr)
+		require.Equal(t, int64(4), userLimits.ActiveUserCount)
+
+		// now we'll delete the user
+		_ = th.App.PermanentDeleteUser(th.Context, newUser)
```
Comment (larkox) [inline server/channels/app/limits_test.go:41]: It is nice to test with permanent deletion, but the normal way to "delete users" is just deactivating them. We should test for that too.

File: server/channels/app/limits_test.go [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+		// now we create a new bot
+		newBot := th.CreateBot()
+
+		userLimits, appErr = th.App.GetUserLimits()
+		require.Nil(t, appErr)
+		require.Equal(t, int64(3), userLimits.ActiveUserCount)
+
+		// now we'll delete the bot
+		_ = th.App.PermanentDeleteBot(newBot.UserId)
```
Comment (larkox) [inline server/channels/app/limits_test.go:60]: Same thing about deactivation.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Only my first comment worries me. The rest are just nits.

File: server/channels/app/limits_test.go (hunk 7 above)
Comment (harshilsharma63) [reply server/channels/app/limits_test.go:119]: Done.

File: server/channels/api4/limits.go (hunk 1 above)
Comment (harshilsharma63) [reply server/channels/api4/limits.go:19]: I think its fine as there is no logic in API layer other than checking for admin role. We can add later as logic grows.

File: server/channels/app/limits.go (hunk 2 above)
Comment (harshilsharma63) [reply server/channels/app/limits.go]: Yes, correct, `maxUsersLimit` check should be the first thing. Fixed and simplified.

File: server/channels/app/limits_test.go (hunk 4 above)
Comment (harshilsharma63) [reply server/channels/app/limits_test.go:31]: Yeah, fine for now.

File: server/channels/app/limits_test.go (hunk 5 above)
Comment (harshilsharma63) [reply server/channels/app/limits_test.go]: Done, added.

File: server/channels/app/limits_test.go (hunk 6 above)
Comment (harshilsharma63) [reply server/channels/app/limits_test.go:41]: Good catch! Done, added,

File: server/channels/app/limits_test.go (hunk 7 above)
Comment (harshilsharma63) [reply server/channels/app/limits_test.go:119]: Done.

**Review by larkox — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Pull request merged**